### PR TITLE
[New Feature][Solace Integration-02] Introduce new gatewayVendor property to APIs

### PR DESCRIPTION
**Pull request by Chamindu36**
### Purpose
Add new property of GatewayVendor to API and APIProducts to support external gateways. Adding the property column into DB tables and support the property search in the persistence layer.

### Goal
Fixes : 
https://github.com/wso2/product-apim/issues/11650

### Unit Tests

- testGetAPIGatewayVendorByUUID - Test DB call of getting the gatewayVendor details when UUID is provided.

### Related PRs 
Marge after the following PR - 
https://github.com/wso2/carbon-apimgt/pull/10839

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java [hunk 1]
```diff
@@ -4047,4 +4048,43 @@ protected boolean isOauthAppValidation() {
...
+    /**
+     * Returns the AsyncAPI definition of the given API
+     *
+     * @param apiId id of the APIIdentifier
+     * @return A String containing the AsyncAPI definition
+     * @throws APIManagementException
+     */
+    @Override
+    public String getAsyncAPIDefinition(Identifier apiId) throws APIManagementException {
+
```
Comment (dnwick) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java:4060]: remove new line

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java [hunk 2]
```diff
@@ -4047,4 +4048,43 @@ protected boolean isOauthAppValidation() {
...
+            //Tenant store anonymous mode if current tenant and the required tenant is not matching
+            if (this.tenantDomain == null || isTenantDomainNotMatching(apiTenantDomain)) {
+                if (apiTenantDomain != null) {
+                    startTenantFlow(apiTenantDomain);
+                    tenantFlowStarted = true;
+                }
+                int tenantId = getTenantManager().getTenantId(apiTenantDomain);
+                registryType = getRegistryService().getGovernanceUserRegistry(
+                        CarbonConstants.REGISTRY_ANONNYMOUS_USERNAME, tenantId
+                );
```
Comment (dnwick) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java:4075]: fix formatting in all places . get to one line if posssible

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java [hunk 3]
```diff
@@ -8067,6 +8069,41 @@ public String getOrganizationByAPIUUID(String apiId) throws APIManagementExcepti
...
+                connection.setAutoCommit(false);
+                ps.setString(1, apiId);
+                result = ps.executeQuery();
+
+                while (result.next()) {
+                    organization = result.getString("GATEWAY_VENDOR");
+                }
+                connection.commit();
+            } catch (SQLException e) {
+                connection.rollback();
```
Comment (dnwick) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java:8095]: what does connection.rollback(); does differntly than calling APIMgtDBUtil.rollbackConnection in the next line ?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java [hunk 4]
```diff
@@ -8067,6 +8069,41 @@ public String getOrganizationByAPIUUID(String apiId) throws APIManagementExcepti
...
+                APIMgtDBUtil.rollbackConnection(connection, "Failed to rollback while fetching gateway vendor" +
+                        " of the API", e);
+            } finally {
+                APIMgtDBUtil.setAutoCommit(connection, initialAutoCommit);
+            }
+        } catch (SQLException e) {
+            handleException("Error occurred while fetching gateway vendor of the API", e);
```
Comment (dnwick) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java:8102]: if we have more information of the API then we can add here

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SQLConstants.java [hunk 5]
```diff
@@ -1417,8 +1417,8 @@
 
     public static final String ADD_API_SQL =
             " INSERT INTO AM_API (API_PROVIDER,API_NAME,API_VERSION,CONTEXT,CONTEXT_TEMPLATE,CREATED_BY," +
-            "CREATED_TIME, API_TIER, API_TYPE, API_UUID,STATUS, ORGANIZATION)" +
-            " VALUES (?,?,?,?,?,?,?,?,?,?,?,?)";
+                    "CREATED_TIME, API_TIER, API_TYPE, API_UUID,STATUS, ORGANIZATION, GATEWAY_VENDOR)" +
```
Comment (dnwick) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SQLConstants.java:1420]: format properly

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java (hunk 1 above)
Comment (Chamindu36) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java]: Resolved with c22511fa4eae6b2a7a040f9077c796b385d6334d

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java (hunk 2 above)
Comment (Chamindu36) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java]: Resolved with c22511fa4eae6b2a7a040f9077c796b385d6334d

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SQLConstants.java (hunk 5 above)
Comment (Chamindu36) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SQLConstants.java]: Resolved with c22511fa4eae6b2a7a040f9077c796b385d6334d

**Review by dnwick — APPROVED** [review]

**Review by tharindu1st — APPROVED** [review]

**Pull request merged**